created chrysom readme draft (#197)
xmidt-org/argus: chrysom/README.md

## Summary
This package enables CRUD operations on the items stored in Argus  The items being stored are valid JSON objects. The package also provides a listener that is able to poll from Argus on an interval.

## CRUD Operations

- On a `GetItems()` call, chrysom returns a list of all items belonging to the provided owner.
- `PushItem()` updates an item if the item exists and the ownership matches. If the item does not exist then a new item will be created. It will return an error or a string saying whether the item was successfully created or updated.
- `RemoveItem()` removes the item if it exists and returns the data associated to it.

## Listener
The client contains a listener that will listen for item updates from Argus on an interval based on the client configuration. 

Listener provides a mechanism to fetch a copy of all items within a bucket on an interval. If not provided, listening won't be enable for this client.

Start begins listening for updates on an interval. A Listener must be given in the ListenerConfig for this to work. If a listener process is already in progress, calling Start() is a NoOp. If you want to restart the current listener process, call Stop() first.

> Stop requests the current listener process to stop and waits for its goroutine to complete. Calling Stop() when a listener is not running (or while one is getting stopped) returns an  error.

Reviewer: Lines 20 and 21 are currently part of the same paragraph, which I think isn't your intention?  You can see by clicking on the button to "display the rich diff" on the files page of the pr or viewing this file in the browser directly:
https://github.com/xmidt-org/argus/blob/b4adcf3e4ef7133e3eeb85a66ebb1f33f9176cae/chrysom/README.md

In markdown, two paragraphs need to have an empty line between them or some other format indicator (bullet point paragraphs can be next to each other because the bullet point makes it clear when we've moved onto the next).

This sounds like a strange feature, but it's helpful if you like keeping your line widths at a max of 80 (which I do).  Our team doesn't have that requirement, so you don't have to change anything, this is just my explanation of how this is helpful. 🙂

# chrysom
The client library for communicating with Argus over HTTP.

> [![PkgGoDev](https://pkg.go.dev/github.com/xmidt-org/argus@v0.5.1/chrysom)](https://pkg.go.dev/github.com/xmidt-org/argus@v0.5.1/chrysom)

Reviewer: ```suggestion

[![Go Reference](https://pkg.go.dev/badge/github.com/xmidt-org/argus/chrysom.svg)](https://pkg.go.dev/github.com/xmidt-org/argus/chrysom)
```

The badge looked broken...I tried to manually fix things, but by all accounts what you did looks right.  I got this suggestion from this page:
https://pkg.go.dev/badge/

I took out the version so we won't have to update this badge when we make a new release.

## Summary

> This package enables CRUD operations with Argus: 

Reviewer: Can you add a bit more to this? It should be like a one to two sentence summary of chrysom that would make sense even if it was copied and pasted outside of this readme.

> ### CRUD Operations

Reviewer: ```suggestion
## CRUD Operations
```

I don't think this is meant to be part of the summary, so we can make it a larger header. 👍

> Gets all items that belong to a given owner
> ```
> func (c *Client) GetItems(ctx context.Context, owner string) (Items, error)
> ```

Reviewer: I would suggest changing each of these to a bullet in the list and then describe what happens when the function is called:

```suggestion
- On a `GetItems()` call, chrysom returns a list of all items belonging to the provided owner.
```

In general, we want our readme to have full sentences.

Updates an item if the item exists and the ownership matches. If the item does not exist then a new item will be created
```
func (c *Client) PushItem(ctx context.Context, owner string, item model.Item) (PushResult, error)
```
Removes the item if it exists and returns the data associated to it.
```
func (c *Client) RemoveItem(ctx context.Context, id, owner string) (model.Item, error)
```

> ### Listener

Reviewer: ```suggestion
## Listener
```

The client contains a listener that will listen for item updates from Argus on an interval based on the client configuration. 

> ``` PullInterval```

Reviewer: I'm not sure what this is here for? But it's inverting/breaking the code snippets after this.

Listener provides a mechanism to fetch a copy of all items within a bucket on an interval. If not provided, listening won't be enable for this client.

> Start begins listening for updates on an interval given that client configuration is setup correctly. If a listener process is already in progress, calling Start() is a NoOp. If you want to restart the current listener process, call Stop() first.

Reviewer: What does it mean to have client configuration set up correctly for listening?

Author: added detail on the configuration required to make the listener function

> ```
> func (c *Client) Start(ctx context.Context) error
> ```

Reviewer: ```suggestion

```

I would also remove these functions here.  I think these Start and Stop paragraphs look good and stand up well on their own. 👍 They don't need the function definitions.

Stop requests the current listener process to stop and waits for its goroutine to complete. Calling Stop() when a listener is not running (or while one is getting stopped) returns an  error.

> ```
> func (c *Client) Stop(ctx context.Context) error
> ```

Reviewer: ```suggestion
```
